### PR TITLE
Ensure functions added to network executor are called

### PR DESCRIPTION
**Pull request by aaronbuchwald**

### CODE REVIEW

File: utils/timer/executor.go [hunk 1]
```diff
@@ -9,11 +9,12 @@ import (
 
 // Executor ...
 type Executor struct {
```
Comment (StephenButtolph) [inline utils/timer/executor.go:11]: It's not clear to me what the changes in this file are meant to do. What's the purpose of this change?

**Review by StephenButtolph — CHANGES_REQUESTED** [review]

File: utils/timer/executor.go (hunk 1 above)
Comment (aaronbuchwald) [reply utils/timer/executor.go:11]: It's meant to separate the lock held during dispatch and the lock held to add an event. So these changes are meant to make it so that calling Add won't block even if a function being executed within the Dispatch goroutine is currently blocking.

Comment (StephenButtolph) [reply utils/timer/executor.go:11]: Calling Add doesn't block when a function is being executed in the current code.

Comment (aaronbuchwald) [reply utils/timer/executor.go]: You're right, just reverted the changes

**Review by StephenButtolph — APPROVED** [review]
